packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeProvider.cs

```diff
@@ -333,6 +424,429 @@ private MethodProvider BuildEmptyConstructor()
                 this);
         }
 
+        /// <summary>
+        /// Produces the validation body statements for the JSON serialization format.
+        /// </summary>
+        private MethodBodyStatement CreateValidateJsonFormat(CSharpType modelInterface, SerializationFormatValidationType validationType)
+        {
+            /*
+                var format = options.Format == "W" ? GetFormatFromOptions(options) : options.Format;
+                if (format != <formatValue>)
+                {
+                    throw new FormatException($"The model {nameof(ThisModel)} does not support '{format}' format.");
+                }
+            */
+            MethodBodyStatement[] statements =
+            [
+                GetConcreteFormat(_mrwOptionsParameterSnippet, modelInterface, out VariableExpression format),
+                new IfStatement(NotEqual(format, ModelReaderWriterOptionsSnippet.JsonFormat))
+                {
+                    ThrowValidationFailException(format, modelInterface.Arguments[0], validationType)
+                },
+            ];
+
+            return statements;
+        }
+
+        private MethodBodyStatement GetConcreteFormat(ModelReaderWriterOptionsSnippet options, CSharpType iModelTInterface, out VariableExpression format)
+        {
+            var castSnippet = new StringSnippet(This.CastTo(iModelTInterface).Invoke(nameof(IPersistableModel<object>.GetFormatFromOptions), options));
+            var condition = new TernaryConditionalExpression(
+                Equal(options.Format, ModelReaderWriterOptionsSnippet.WireFormat),
+                castSnippet,
+                options.Format);
+            var reference = new VariableExpression(castSnippet.Type, "format");
+            format = reference;
+            return Var(reference, condition);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="KeywordStatement"/> of type <see cref="FormatException"/> with a specific message indicating
```

Comment:
nit: when the method is self explanatory I would avoid this kind of documentation on private methods.